src/System.Private.Reflection.Core/src/System/Reflection/Runtime/Assemblies/EcmaFormat/EcmaFormatRuntimeAssembly.cs

```diff
@@ -33,10 +33,25 @@ namespace System.Reflection.Runtime.Assemblies.EcmaFormat
 {
     internal sealed partial class EcmaFormatRuntimeAssembly : RuntimeAssembly
     {
-        private EcmaFormatRuntimeAssembly(MetadataReader reader)
+        public override string Location { get; }
+
+        public override string CodeBase { get; }
+
+        private EcmaFormatRuntimeAssembly(MetadataReader reader, string assemblyPath)
         {
             AssemblyDefinition = reader.GetAssemblyDefinition();
             MetadataReader = reader;
+
+            if (assemblyPath != null)
+            {
+                Location = assemblyPath;
+                CodeBase = "file:///" + assemblyPath.Replace('\\', '/').TrimStart('/');
+            }
+            else
+            {
+                Location = "";
+                CodeBase = "file:///" + Path.Combine(AppContext.BaseDirectory, typeof(object).Assembly.ManifestModule.Name).Replace('\\', '/').TrimStart('/');
```

Comment:
The best effort simulation of CodeBase can be done in the base class. It is not really Ecma assembly specific.